src/components/FieldsetLegend/FieldsetLegend.tsx

```diff
@@ -0,0 +1,43 @@
+import clsx from 'clsx';
+import React from 'react';
+import styles from './FieldsetLegend.module.css';
+
+export interface FieldsetLegendProps {
+  /**
+   * CSS class names that can be appended to the component.
+   */
+  className?: string;
+  /**
+   * String for the optional legend. By default it is '(optional)'.
+   */
+  optionalLabel?: string;
+  /**
+   * Indicates that field is required for form to be successfully submitted.
+   */
+  required?: boolean;
+  /**
+   * Legend text string that names the fieldset.
+   */
+  text: string;
+}
+
+/**
+ * Primary UI component for user interaction
+ */
+export const FieldsetLegend = ({
+  className,
+  optionalLabel = '(optional)',
+  text,
+  required = true,
+  ...other
+}: FieldsetLegendProps) => {
+  const componentClassName = clsx(styles['legend'], className);
+  return (
+    <legend className={componentClassName} {...other}>
+      {text}{' '}
+      {!required && (
+        <span className={styles['legend__flag']}>{optionalLabel}</span>
+      )}
```

Comment:
This is interesting. It's assumed that all form groups are required unless specified otherwise? I think in the product they do it the other way – assume all controls are optional unless marked as required. But that's for individual controls; I'm not used to seeing form *groups* labeled as required or optional.

Reply:
Hmm thanks for bringing up this consideration.
Since the `required` prop is used for presentation of this `optionalLabel` only, will refactor to remove the `required` prop and make the `optionalLabel` (or rename more appropriately) an optional enum between `(required)` or `(optional)`.
Let me know if you think there's value in keeping the `required` label in case of using a red asterisk to indicate required status, or alternatively the `optionalLabel` enum should be `(required)`, `(optional)`, and `*`

Reply:
Ok, how do you feel about https://github.com/chanzuckerberg/edu-design-system/pull/971/files#diff-2fbe26d11df22cb350e98e747e952b42398d0b34fe51b333efcbc6fa2adf3d16R13:

![image](https://user-images.githubusercontent.com/86632227/163455741-386f7e76-1aae-4618-b76d-7e46a34e2f4d.png)
![image](https://user-images.githubusercontent.com/86632227/163455762-86cb9bf1-5cff-45f1-a4c4-d2299bc9a4e8.png)